File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer,Integer> hm = new HashMap<>();

        for(int i=0;i<nums.length;i++){

            hm.put(nums[i],hm.getOrDefault(nums[i],i));

            if (hm.containsKey(target - nums[i]) && i != hm.get(target - nums[i])) {
                return new int[]{i, hm.get(target - nums[i])};
            }

        }

        return new int[]{0,0};
    }
}

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map = {}

        for i, num in enumerate(nums):
            if target - num in map.keys():
                return [map[target-num], i]
            else:
                map[num] = i

        return [0,0]